test/integration/short/client-execute-prepared-tests.js

```diff
@@ -12,8 +12,117 @@ var vit = helper.vit;
 var vdescribe = helper.vdescribe;
 var Uuid = types.Uuid;
 
+var SimulacronCluster = require('../simulacron-helper');
+
 describe('Client', function () {
   this.timeout(120000);
+  describe('Preparing statements on nodes behavior', function () {
+    var sCluster = null;
+    var client = null;
+    var query = util.format('SELECT * FROM ks.table1 WHERE id1 = ?');
+    beforeEach(function (done) {
+      utils.series(
+        [
+          function stopAnyCcmCluster(next) {
+            helper.ccmHelper.removeIfAny(next);
+          },
+          function startCluster(next) {
+            sCluster = new SimulacronCluster("5", "3.10", "", "test", true, 1, function (err) {
```

Comment:
We should avoid callbacks in constructors, as you can't get the context of the current instance.

```js
var myVar = new MyConstructor(function() {
  // I can't access myVar here (it depends on implementation of MyConstructor)
});
```
vs
```js
var myVar = new MyConstructor();
myVar.method(function () {
  // I can access myVar
});
```